common/OpTestHMC.py

```diff
@@ -307,17 +307,97 @@ def get_lpar_cfg(self):
             cfg_dict[key] = value
         return cfg_dict
 
-    def set_lpar_cfg(self, arg_str):
+    def set_lpar_cfg(self, arg_str, lpar_profile = None):
         '''
         Set LPAR configuration parameter values
 
         :param arg_str: configuration values in key, value pair seperated by
                         comma
         '''
+        lpar_profile = self.lpar_prof if lpar_profile is None else lpar_profile
         if not self.lpar_prof:
             raise OpTestError("Profile needs to be defined to use this method")
         self.ssh.run_command("chsyscfg -r prof -m %s -i 'lpar_name=%s,name=%s,%s' --force" %
-                (self.mg_system, self.lpar_name, self.lpar_prof,arg_str))
+                (self.mg_system, self.lpar_name, lpar_profile, arg_str))
+
+    def add_ioslot(self, add_ioslot, lpar_profile = None):
+        """
+        Adds ioslots to lpar profile
+        :param add_ioslot: String, accepts drc name or drc names
+                           seperated by comma(,)
+        drc name example: U780C.ND0.WZS0042-P1-C2
+        :param lpar_profile: String, Defaults to class variable
+                             if not provided
+        """
+        lpar_profile = self.lpar_prof if lpar_profile is None else lpar_profile
+        exisitng_io_slots = self.get_ioslots_assigned_to_lpar(lpar_profile = lpar_profile)
+        drc_index_for_slots = [self.get_drc_index_for_ioslot(add_slot)
+                               for add_slot in add_ioslot.split(",")]
+        if None in drc_index_for_slots:
```

Comment:
just  `if not drc_index_for_slots`  will work

Reply:
@abdhaleegit drc_index_for_slots is an array. Trying to catch if any of the value os "None". Let me know if I need to correc it still